libgit/autogit_node_wrappers.go

```diff
@@ -430,32 +176,68 @@ func (arn autogitRootNode) WrapChild(child libkbfs.Node) libkbfs.Node {
 type rootNode struct {
 	libkbfs.Node
 	am *AutogitManager
+
+	lock sync.RWMutex
+	fs   *libfs.FS
 }
 
 var _ libkbfs.Node = (*rootNode)(nil)
 
 // ShouldCreateMissedLookup implements the Node interface for
 // rootNode.
-func (rn rootNode) ShouldCreateMissedLookup(ctx context.Context, name string) (
+func (rn *rootNode) ShouldCreateMissedLookup(ctx context.Context, name string) (
 	bool, context.Context, libkbfs.EntryType, string) {
-	if name == autogitRoot {
-		ctx = context.WithValue(ctx, libkbfs.CtxReadWriteKey, struct{}{})
-		ctx = context.WithValue(ctx, libkbfs.CtxAllowNameKey, autogitRoot)
-		return true, ctx, libkbfs.Dir, ""
+	if name != AutogitRoot {
+		return rn.Node.ShouldCreateMissedLookup(ctx, name)
+	}
+
+	rn.lock.Lock()
+	defer rn.lock.Unlock()
+	if rn.fs == nil {
+		// Make the FS once, in a place where we know the NodeCache
+		// won't be locked (to avoid deadlock).
+
+		// Wrap this child so that it will show all the
+		// repos. TODO(KBFS-3429): fill in context.
+		ctx := context.TODO()
+		h, err := rn.am.config.KBFSOps().GetTLFHandle(ctx, rn)
```

Comment:
Perhaps this context should be different from what's passed into `NewFS` below, so it can have a timeout.

Reply:
You're right, I should be using the passed-in `ctx` here. Will fix.